test/versioning-strategies/prerelease.ts

```diff
@@ -0,0 +1,280 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it} from 'mocha';
+
+import {expect} from 'chai';
+import {PrereleaseVersioningStrategy} from '../../src/versioning-strategies/prerelease';
+import {Version} from '../../src/version';
+
+describe('PrereleaseVersioningStrategyVersioningStrategy', () => {
```

Comment:
Also add test for:

- 1.0.0-beta1 => 1.0.0-beta2
- 1.0.0-beta01 => 1.0.0-beta02
- 1.0.0-beta9 => 1.0.0-beta10 (although that would be unfortunate)
- 1.0.0-beta01 => 1.0.0-beta10

Reply:
> * 1.0.0-beta01 => 1.0.0-beta10

I think this one was intended to be `1.0.0-beta09 => 1.0.0-beta10`

Reply:
Yup, absolutely. I was clearly not at my best when commenting. Sorry :(